tests/MassTransit.Metrics.Tests/Metrics_Specs.cs

```diff
@@ -0,0 +1,95 @@
+namespace MassTransit.Metrics.Tests
+{
+    using MassTransit.Metrics.Tests.HarnessContracts;
+    using MassTransit.Monitoring;
+    using MassTransit.Testing;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Diagnostics.Metrics.Testing;
+    using NUnit.Framework;
+    using System.Threading.Tasks;
+
+    namespace HarnessContracts
+    {
+        using System;
+
+        public interface SubmitOrder
+        {
+            Guid OrderId { get; }
+            string OrderNumber { get; }
+        }
+
+
+        public interface OrderSubmitted
+        {
+            Guid OrderId { get; }
+            string OrderNumber { get; }
+        }
+    }
+
+
+    [TestFixture]
+    [Explicit]
+    public class Metrics_Specs
+    {
+        [Test]
+        public async Task Should_get_metrics_with_tags()
+        {
+            var collector = new MetricCollector<long>(null, InstrumentationOptions.MeterName, "messaging.masstransit.consume");
```

Comment:
don't you need to dispose it?

Reply:
and another thing, you can retrieve metric name from `InstrumentationOptions` options

Reply:
I’ll check it. If it is disposable then it should be most probably disposed.

Reply:
> and another thing, you can retrieve metric name from `InstrumentationOptions` options

I wouldn’t rely on too many external classes for assert. I would rather use some kind of a snapshot testing using Verify.NUnit and a json verification file.

Reply:
I'm not sure what you mean, but MetricName for collector is defined in InstrumentationOptions, so instead of constant metric name I'd use one in options.